src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadata.scala

```diff
@@ -93,7 +93,21 @@ case class DatasetMetadata(private val identifiers: Option[Seq[SchemedValue]] =
 
   /** Validations as far as not covered by DDM schema validation. */
   private[docs] def validate(): Try[Unit] = {
-    Author.validate(authors)
+    def missingMandatory[T <: Mandatory](values: Seq[T]): Try[Unit] = {
+      val invalid = values
+        .withFilter(!_.hasMandatory)
+        .map(x => x.getClass.getSimpleName + JsonUtil.toJson(x))
+        .mkString(", ")
+      if (invalid.isEmpty) Success(())
+      else {
+        Failure(new IllegalArgumentException(s"Missing mandatory values in: $invalid"))
+      }
+    }
+
+    for { // TODO collect errors over multiple types of validation errors
```

Comment:
you are looking for https://typelevel.org/cats/datatypes/validated.html 🤔 
we'll get to that later 😜 
keep the TODO for now!